src/torrent.cpp

```diff
@@ -821,6 +841,20 @@ std::vector<std::shared_ptr<Peer>> Torrent::tracker_request(
     std::rethrow_exception(thrown);
   }
 
+  // To avoid connecting to our own listening peer
+  const auto is_local_peer = [&](const auto& peer) {
+    const auto& url = peer->url();
+    // FIXME: Get hold of the ip automatically?
+    return url &&
+           (url->host() == "localhost" || url->host() == "172.17.0.1" ||
+            url->host() == "192.168.0.18") &&
```

Comment:
192.168.0.18 should not be hardcoded, possibly the docker range too? Let it be configurable.